geotiepoints/_modis_interpolator.pyx

```diff
@@ -582,32 +585,108 @@ cdef class MODISInterpolator:
-                            expanded_arr[row_offset + length_repeat_cycle,
-                                        col_offset + factor + factor + width_repeat_cycle] = tiepoint_value
+                self._expand_tiepoint_array_5km_with_repeat(tiepoint_value, expanded_arr, row_offset, col_offset)
+
+    @cython.boundscheck(False)
+    @cython.cdivision(True)
+    @cython.wraparound(False)
+    @cython.initializedcheck(False)
+    cdef void _expand_tiepoint_array_5km_left(self, floating[:, :] input_arr, floating[:, ::1] expanded_arr) nogil:
+        cdef floating tiepoint_value
+        cdef Py_ssize_t row_idx, col_idx, row_offset, col_offset
+        col_idx = 0
+        col_offset = col_idx * self._fine_pixels_per_coarse_pixel
+        for row_idx in range(input_arr.shape[0]):
+            row_offset = row_idx * self._fine_pixels_per_coarse_pixel * 2
+            tiepoint_value = input_arr[row_idx, col_idx]
+            self._expand_tiepoint_array_5km_with_factor_repeat(tiepoint_value, expanded_arr, row_offset, col_offset)
+
+    @cython.boundscheck(False)
+    @cython.cdivision(True)
+    @cython.wraparound(False)
+    @cython.initializedcheck(False)
+    cdef void _expand_tiepoint_array_5km_270_extra_column(
+            self,
+            floating[:, :] input_arr,
+            floating[:, ::1] expanded_arr
+    ) nogil:
+        """Copy an extra coarse pixel column between the main copied area and the right-most columns."""
+        cdef floating tiepoint_value
+        cdef Py_ssize_t row_idx, col_idx, row_offset, col_offset
+        col_idx = input_arr.shape[1] - 1
+        col_offset = col_idx * self._fine_pixels_per_coarse_pixel + self._fine_pixels_per_coarse_pixel + self._factor_5km
+        for row_idx in range(input_arr.shape[0]):
+            row_offset = row_idx * self._fine_pixels_per_coarse_pixel * 2
+            tiepoint_value = input_arr[row_idx, col_idx]
+            self._expand_tiepoint_array_5km_with_repeat(tiepoint_value, expanded_arr, row_offset, col_offset)
+
+    @cython.boundscheck(False)
+    @cython.cdivision(True)
+    @cython.wraparound(False)
+    @cython.initializedcheck(False)
+    cdef void _expand_tiepoint_array_5km_right(self, floating[:, :] input_arr, floating[:, ::1] expanded_arr) nogil:
+        """Add the right most expanded columns.
+        
+        This is always the right-most "factor"s-worth of columns.
+        
+        """
+        cdef floating tiepoint_value
+        cdef Py_ssize_t row_idx, col_idx, row_offset, col_offset
+        col_idx = input_arr.shape[1] - 1
+        col_offset = expanded_arr.shape[1] - self._factor_5km
+        for row_idx in range(input_arr.shape[0]):
+            row_offset = row_idx * self._fine_pixels_per_coarse_pixel * 2
+            tiepoint_value = input_arr[row_idx, col_idx]
+            self._expand_tiepoint_array_5km_with_factor_repeat(tiepoint_value, expanded_arr, row_offset, col_offset)
```

Comment:
This code is duplicated from `_expand_tiepoint_array_5km_left`, please refactor

Reply:
Done.